### PR TITLE
Test/localdescriptors

### PR DESCRIPTION
**Pull request by vyasr**
Adds tests of LocalDescriptors, including one that can be used to verify ref_points vs points behavior.

### CODE REVIEW

File: tests/util.py
Comment (bdice) [inline tests/util.py]: Probably should be `4*nx*ny*nz` but maybe there's a better way to write that across all of these functions.

File: tests/util.py
Comment (bdice) [inline tests/util.py]: SC ➡️ simple cubic; add period.

File: tests/util.py
Comment (bdice) [inline tests/util.py]: Be consistent with how the numbers are given. `0.5` vs `.5`

File: tests/util.py
Comment (bdice) [inline tests/util.py]: Shape as above.

File: tests/util.py
Comment (bdice) [inline tests/util.py]: body-centered cubic, add period.

File: tests/util.py
Comment (bdice) [inline tests/util.py]: face-centered cubic; add period.

File: tests/util.py
Comment (bdice) [inline tests/util.py]: Missing colon before `:py:class:`; inconsistent with use of `:py:class:` vs. `:class:` (assumed Python namespace).

File: tests/test_environment_LocalDescriptors.py
Comment (bdice) [inline tests/test_environment_LocalDescriptors.py]: Missing a close parenthesis )

File: tests/test_environment_LocalDescriptors.py
Comment (bdice) [inline tests/test_environment_LocalDescriptors.py:152]: Let's use a NeighborQuery API instead, so that we won't have to refactor this later.

File: tests/test_environment_LocalDescriptors.py
Comment (bdice) [inline tests/test_environment_LocalDescriptors.py]: Why make this a separate class?

File: tests/test_environment_LocalDescriptors.py
Comment (bdice) [inline tests/test_environment_LocalDescriptors.py:182]: Use NeighborQuery API

**Review by bdice — CHANGES_REQUESTED** [review]
Good to add these tests! I left some comments below.

File: tests/util.py
Comment (vyasr) [reply tests/util.py]: It's fine, I'll just add it this way. It's a util helper, so more important to have the information than have it in a way that renders nicely.

File: tests/test_environment_LocalDescriptors.py
Comment (vyasr) [reply tests/test_environment_LocalDescriptors.py:152]: Good call, I was thinking that we shouldn't use the new API on master but should be fine.

File: tests/test_environment_LocalDescriptors.py
Comment (vyasr) [reply tests/test_environment_LocalDescriptors.py]: That was just for easily running just the tests I wanted. I'll remove.

**Comment by vyasr** [comment]
I actually decided to hold off changing the API to use the new one for the NearestNeighbors. The reason is the (known) issue that the ordering of indices on master is still in the (ref_point, point) order, and for NearestNeighbors the results aren't symmetric. Once we merge #342, everything on the next branch will have a consistent ordering, and we'll get the expected results passing the data through, but at present, switching to the new API here would require also reversing the neighbor list indices, and we'd have to undo that later anyway. @bdice let me know if you're OK with that.

**Review by bdice — APPROVED** [review]
Looks good to me, after explaining the choice of older neighbor APIs.

**Pull request merged**
